src/desktop_notifier/dbus.py

```diff
@@ -187,28 +154,28 @@ async def _send(
         # raise an AttributeError if required.
         platform_nid = await self.interface.call_notify(  # type:ignore[attr-defined]
             self.app_name,
-            replaces_nid,
+            0,
             icon,
             notification.title,
             notification.message,
             actions,
             hints,
             timeout,
         )
+        self._platform_nid_to_identifier[platform_nid] = notification.identifier
 
-        notification.identifier = identifier_from_dbus(platform_nid)
-
-    async def _clear(self, notification: Notification) -> None:
+    async def _clear(self, identifier: str) -> None:
         """
         Asynchronously removes a notification from the notification center
         """
         if not self.interface:
             return
 
+        platform_nid = self._platform_nid_to_identifier.inverse.pop(identifier)
```

Comment:
minor but is it a good idea to `.pop()` before verifying that the call to `close_notification()` has succeeded? seems like it might be wise to wait for confirmation that the notification has been closed before we stop tracking it instead of other way around.

Reply:
DBusError

Reply:
Good point, done.